server/src/main/java/org/elasticsearch/action/DelegatingActionListener.java

```diff
@@ -11,8 +11,10 @@
 import static org.elasticsearch.action.ActionListenerImplementations.safeOnFailure;
 
 /**
- * A wrapper around an {@link ActionListener} which delegates failures safely to the inner listener's {@link ActionListener#onFailure}
- * method and which has a {@link #toString()} implementation which describes this class and the delegate.
+ * A wrapper around an {@link ActionListener} {@code L} that delegates failures to {@code L}'s {@link ActionListener#onFailure} method.
```

Comment:
Several subclasses override the `onFailure` behaviour too, it's only a useful default.

```suggestion
 * A wrapper around an {@link ActionListener} {@code L} that by default delegates failures to {@code L}'s {@link ActionListener#onFailure} method.
```

In some sense the "real" reason for this class is more that it has a `toString()` implementation that includes `delegate.toString()` which makes these things less useless when they appear in logs.

Reply:
Returned the toString() comment. I thought of it more as an implementation detail, rather than something a user needs to know, but if it's helpful.

I'm hesitant about 'by default' because the user of this class cannot provide any other delegation. I added a line about overriding other methods.

Reply:
> the user of this class cannot provide any other delegation

Perhaps I'm misunderstanding what you mean by "user" but there's quite a few implementations that do something different from just delegating failures to `L.onFailure()`, see for instance `org.elasticsearch.action.ActionListenerImplementations.DelegatingResponseActionListener` or `org.elasticsearch.action.support.CountDownActionListener`.

Reply:
Ah, I think I was missing that this is an abstract class, so there are no uses of DelegatingActionListener.

I was think of "user" as any code with `new DelegatingActionListener`. But that doesn't exist :)

Done.